org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/editor/MavenTargetLocationWizard.java

```diff
@@ -95,14 +93,63 @@ public String getText(Object element) {
 					type.setText(targetLocation.getArtifactType());
 					scope.setText(targetLocation.getDependencyScope());
 					metadata.setSelection(new StructuredSelection(targetLocation.getMetadataMode()));
+					bndInstructions = targetLocation.getInstructions(null);
 				} else {
 					artifactId.setText("");
 					groupId.setText("");
 					version.setText("");
 					type.setText(MavenTargetLocation.DEFAULT_PACKAGE_TYPE);
 					scope.setText(MavenTargetLocation.DEFAULT_DEPENDENCY_SCOPE);
 					metadata.setSelection(new StructuredSelection(MavenTargetLocation.DEFAULT_METADATA_MODE));
+					bndInstructions = new BNDInstructions("", null);
 				}
+				updateButton();
+			}
+
+			private void createMetadataCombo(Composite parent) {
+				Composite composite = new Composite(parent, SWT.NONE);
+				composite.setLayout(new GridLayout(2, false));
+				metadata = new ComboViewer(composite);
+				metadata.setContentProvider(ArrayContentProvider.getInstance());
+				metadata.setLabelProvider(new ColumnLabelProvider() {
+					@Override
+					public String getText(Object element) {
+						if (element instanceof MissingMetadataMode) {
+							return ((MissingMetadataMode) element).name().toLowerCase();
+						}
+						return super.getText(element);
+					}
+
+				});
+				editInstructionsButton = new Button(composite, SWT.PUSH);
```

Comment:
IMO, this would be better be a link, as the button besides the combo looks heavy and -unless I'm mistaken- the new dialog pops-up on top of the current one and doesn't affect the "parent" dialog. Links IMO better represent such "navigate without changing state" approach.